fetch.bs

```diff
@@ -6224,17 +6224,22 @@ constructor must run these steps:
   <p>If <var>inputBody</var> is <var>body</var> and <var>inputBody</var> is non-null, then:
 
   <ol>
-   <li>
-    <p>Let <var>rs</var> bs a {{ReadableStream}} object from which one can read the exactly
-    same data as one could read from <var>inputBody</var>'s <a for=body>stream</a>.
+   <li><p>Let <var>ws</var> and <var>rs</var> be the
+   <a href="https://streams.spec.whatwg.org/#writable-side">writable side</a> and
+   <a href="https://streams.spec.whatwg.org/#readable-side">readable side</a> of an
+   <a href="https://streams.spec.whatwg.org/#identity-transform-stream">identity transform
+   stream</a>, respectively.</li>
 
-    <p class=XXX>This will be specified more precisely once
-    <a href=https://streams.spec.whatwg.org/#ts-model>transform stream</a> and
-    <a href=https://streams.spec.whatwg.org/#pipe-chains>piping</a> are precisely defined.
-    See <a href=https://github.com/whatwg/fetch/issues/463>issue #463</a>.
+   <li>
+    <p>Let <var>promise</var> be the result of calling
+    <a abstract-op>ReadableStreamPipeTo</a>(<var>inputBody</var>, <var>ws</var></var>, false,
+    false, false, undefined).
 
     <p class="note no-backref">This makes <var>inputBody</var>'s <a for=body>stream</a>
     <a for=ReadableStream>locked</a> and <a for=ReadableStream>disturbed</a> immediately.
+   </li>
+
+   <li><p>Set <var>promise</var>.\[[PromiseIsHandled]]</var> to true.
```

Comment:
Is there a way to avoid this? Manipulating internal state of promises seems problematic and especially doing so in a specification that isn't likely to be notified if ECMAScript is refactored in some fashion.

Reply:
I'm not aware of one other than calling `then` (or `catch`) explicitly. @domenic, do you have any recommendation?

Reply:
In my opinion this is an exceptional internal slot that it is OK for other standards to manipulate. (As an alternative to uselessly calling PerformPromiseThen() with a no-op catch handler.)

Reply:
What do you think of moving this into IDL somehow with other specifications having "**Handle** _promise_." or some such?

Reply:
That seems like a reasonable future cleanup, yeah. As-is right now we're still in the world where streams operates on ES promises directly.